segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -146,7 +147,8 @@ public void setupAppend(SetupAppend setupAppend) {
             try {
                 JsonWebToken token = tokenVerifier.verifyToken(newSegment,
                         setupAppend.getDelegationToken(),
-                        AuthHandler.Permissions.READ_UPDATE);
+                        newSegment.contains(NameUtils.INTERNAL_STREAM_IDENTIFYING_PATTERN) ?
```

Comment:
This feels hacky. Could the delegation token instead contain specific authorization for the internal stream that needs to be written to? This current way is amplifying a read authorization, which I'm not convinced is a good approach.

Reply:
Sure, I've changed the implementation now such that the delegation token is assigned the appropriate authorization. Please see if it looks better.